### PR TITLE
[Exporter.Geneva] Update scopes

### PR DESCRIPTION
**Pull request by utpilla**
### Exporting scopes in GenevaExporter:

1. Export scopes which have a non-null key as individual columns (each key-value pair from the scopes is exported as its own column; these columns would also be taken into consideration when the CustomFields option is applied)
2. GenevaExporter would not check for duplicate keys (if the scopes and log message together have any duplicate keys, they would all be exported)

### Examples

**1. Key is null**
```csharp
using (logger.BeginScope("MyScope"))
{
    logger.LogInformation("Hello from {food} {price}.", "artichoke", 3.99);
}
```
No scopes are exported in this case as there is no key provided for the scope.

**2. Using formatted string in scopes**
```csharp
using (logger.BeginScope("MyInnerInnerScope with {name} and {age} of custom", "John Doe", 35))
{
    logger.LogInformation("Hello from {food} {price}.", "artichoke", 3.99);
}
```
Scopes exported:

|	      name|	age|
|----------:|---:|
|	"John Doe"|	 35|

**3. Using `IEnumerable<KeyValuePair<string, object>>` in scopes**
```csharp
using (logger.BeginScope(new List<KeyValuePair<string, object>>
{
    new KeyValuePair<string, object>("MyKey1", "MyValue1"),
    new KeyValuePair<string, object>("MyKey2", "MyValue2"),
}))
{
    logger.LogInformation("Hello from {food} {price}.", "artichoke", 3.99);
}
```
Scopes exported:

|MyKey1	|MyKey2|
|-----:|------:|
|MyValue1	|MyValue2|

**4. Duplicate Keys in scopes**

```csharp
using (logger.BeginScope(new List<KeyValuePair<string, object>>
{
    new KeyValuePair<string, object>("MyKey1", "MyValue1"),
    new KeyValuePair<string, object>("MyKey1", "MyValue1"),
}))
{
    logger.LogInformation("Hello from {food} {price}.", "artichoke", 3.99);
}
```
Scopes exported:

|MyKey1	|MyKey1|
|-----:|------:|
|MyValue1	|MyValue1|

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Geneva/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 ## Unreleased
 
+* Update export logic for scopes:
+  * Export scopes which have a non-null key as individual columns (each
+    key-value pair from the scopes is exported as its own column; these columns
+    would also be taken into consideration when the CustomFields option is
+    applied)
+  * GenevaExporter would not check for duplicate keys (if the scopes and log
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Geneva/CHANGELOG.md:10]: I'd remove the part about duplicate. Its already the existing behavior, so don't think it should be in changelog.

File: src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs [hunk 2]
```diff
@@ -349,42 +349,27 @@ internal int SerializeLogRecord(LogRecord logRecord)
...
+                if (this.m_customFields == null || this.m_customFields.ContainsKey(scopeItem.Key))
+                {
+                    if (scopeItem.Value != null)
+                    {
+                        // null is not supported.
+                        cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, scopeItem.Key);
+                        cursor = MessagePackSerializer.Serialize(buffer, cursor, scopeItem.Value);
+                        cntFields += 1;
+                    }
+                }
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs:371]: hasEnvProperties = true; --> need this in else part.

File: src/OpenTelemetry.Exporter.Geneva/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+* Update export logic for scopes:
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Geneva/CHANGELOG.md:5]: *breaking* warning would be nice as this is breaking change from previous beta

**Review by cijothomas — APPROVED** [review]

**Comment by utpilla** [comment]
Related to #389

**Pull request merged**